src/pivot_comb.jl

```diff
@@ -1,23 +1,28 @@
 """
-x = pivot_comb(A, b; ...)
+    x = pivot_comb(A, b; ...)
 
 Solves non-negative least-squares problem by block principal pivoting method
 with combinatorial grouping of least-squares problems. Algorithm 2 described
 in Kim & Park (2011).
 
 Optional arguments:
-    tol: tolerance for nonnegativity constraints
-    max_iter: maximum number of iterations
+* `tol`: tolerance for nonnegativity constraints; default: `1e-8`
+* `max_iter`: maximum number of iterations; default: `30*size(A,2)`
+* `P!`: initial guess of passive set; default: `falses(size(A,2), size(B,2))`
+Alert: this optional argument, if provided, is mutated to become the final passive set.
 
 References:
     J. Kim and H. Park, Fast nonnegative matrix factorization: An
     active-set-like method and comparisons, SIAM J. Sci. Comput., 33 (2011),
     pp. 3261–3281.
 """
-function pivot_comb(A,
-               B::AbstractMatrix{T};
-               tol::Float64=1e-8,
-               max_iter=30*size(A,2)) where {T}
+function pivot_comb(
+    A,
+    B::AbstractMatrix{T};
+    tol::Float64=1e-8,
+    max_iter=30*size(A,2),
+    P!::AbstractMatrix{Bool} = falses(size(A,2), size(B,2)),
+) where {T}
 
```

Comment:
Not sure, but do you think that we should test for `size(P!) == (size(A,2),size(B,2))`, and throw an error otherwise?

Reply:
Forget about it, because than we should also check the ’A’ and ’b’s and this is done nowhere ...